prometheus/histogram.go

```diff
@@ -1682,13 +1682,13 @@ func makeNativeExemplars(ttl time.Duration, maxCount int) nativeExemplars {
 }
 
 func (n *nativeExemplars) addExemplar(e *dto.Exemplar) {
+	n.Lock()
+	defer n.Unlock()
```

Comment:
As said in the commit description, let's discuss the optimization of this in #1609.

Reply:
we agreed on the solution in 1609, which is now in here.